core/services/keystore/keys/workflowencryptionkey/key.go

```diff
@@ -0,0 +1,107 @@
+package workflowencryptionkey
+
+import (
+	cryptorand "crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"golang.org/x/crypto/curve25519"
+	"golang.org/x/crypto/nacl/box"
+)
+
+type Raw []byte
+
+func (raw Raw) Key() Key {
+	privKey := [32]byte(raw)
+	return Key{
+		privateKey: &privKey,
+		publicKey:  curve25519PubKeyFromPrivKey(privKey),
+	}
+}
+
+func (raw Raw) String() string {
+	return "<Encryption Raw Private Key>"
+}
+
+func (raw Raw) GoString() string {
+	return raw.String()
+}
+
+func (raw Raw) Bytes() []byte {
+	return ([]byte)(raw)
+}
+
+type Key struct {
+	privateKey *[curve25519.PointSize]byte
+	publicKey  *[curve25519.PointSize]byte
+}
+
+func New() (Key, error) {
+	publicKey, privKey, err := box.GenerateKey(cryptorand.Reader)
```

Comment:
any reason while public stays public, whereas private becomes priv?

Reply:
no reason, will make them pubKey & privKey 👍🏼

Reply:
it is almost privateKey and publicKey everywhere. So, I would propose to spell out the names fully.